Cesium.CodeGen/Ir/Expressions/BinaryOperators/BinaryOperatorExpression.cs

```diff
@@ -173,6 +175,8 @@ public IType GetExpressionType(IDeclarationScope scope)
 
         // both bitwise and arithmetic operators obey same arithmetic conversions
         // https://en.cppreference.com/w/c/language/operator_arithmetic
+        leftType = leftType.EraseConstType();
+        rightType = rightType.EraseConstType();
```

Comment:
Hey, what do you think of moving this inside of `TypeSystem.GetCommonNumericType`?

Otherwise, every client of that class will have to call these to handle `const`.

In particular, currently we have `ConditionalExpression` calling it as well — I bet we'll have problems comparing const and non-const numbers then.

Reply:
Not sure about it. I think `ConstType` as type is bad decision. My current thinking is that `IsConst` shoud be property of all type.

Also it's only 2 places. I want to avoid generalization when possible. I also do not have mental model why const correctness important for casting everywhere, and that logic may, or may not, be related to place where we call this function

Reply:
Alright. I thought about this a bit, and I am not against making constness a property of a type. But in either case, we'll need a set of helper methods to compare/check assignability/get rid of `const`, so I'd not believe it would bring us much value either.

Reply:
Regarding the problem itself — let's leave this for another time when somebody will want to compare a const and a non-const number :)